gitCoreApi/src/main/java/com/virtuslab/gitcore/api/IGitCoreRepository.java

```diff
@@ -22,5 +22,5 @@
 
   Path getMainDirectoryPath();
 
-  boolean isAncestor(BaseGitCoreCommit presumedAncestor, BaseGitCoreCommit presumedDescendant) throws GitCoreException;
+  boolean isAncestor(IGitCoreCommit presumedAncestor, IGitCoreCommit presumedDescendant) throws GitCoreException;
```

Comment:
while we are here, get rid of  `GitCoreException` from `getAllRemoteBranches()` 
(here and within `IGitCoreRepository`)

Reply:
The problem is... `com.virtuslab.gitcore.impl.jgit.GitCoreRepository#getAllRemoteBranches` does throw `GitCoreException`, it's just "sneakified" by `Try.of(...).get()` (actually, pretty much equivalent to `@lombok.SneakyThrows`/`lombok.Lombok.sneakyThrow`)

Reply:
I reworked it now with [`Try.traverse`](https://www.javadoc.io/doc/io.vavr/vavr/0.10.0/io/vavr/control/Try.html#traverse-java.lang.Iterable-java.util.function.Function-), it's more verbose but also clearer/more correct